### PR TITLE
parser: Support spaces everywhere

### PR DESCRIPTION
**Pull request by Gouthamve**
@fabxc @brian-brazil

---
This change is [Reviewable](https://reviewable.io/reviews/prometheus/prometheus/2869)

### CODE REVIEW

**Comment by Gouthamve** [comment]
@brian-brazil @fabxc  This fixes #2852

File: pkg/textparse/parse_test.go [hunk 1]
```diff
@@ -109,14 +139,14 @@ func TestParseErrors(t *testing.T) {
...
 	}{
-		{
-			input: "a",
-			err:   "no token found",
-		},
-		{
-			input: "a{b='c'} 1\n",
-			err:   "no token found",
-		},
+		//{
```
Comment (brian-brazil) [inline pkg/textparse/parse_test.go]: Is this meant to be commented out?

Comment (Gouthamve) [inline pkg/textparse/parse_test.go]: Sorry, crept in by mistake. Fixed it.

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**